control-plane/api/mesh/v2beta1/gateway_class_config_types.go

```diff
@@ -59,19 +59,21 @@ type GatewayClassDeploymentConfig struct {
 	// influence the placement of workloads based on node attributes.
 	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
-	// PriorityClassName specifies the priority class name to use on the created Deployment
+	// PriorityClassName specifies the priority class name to use on the created Deployment.
 	PriorityClassName string `json:"priorityClassName,omitempty"`
-	// Replicas specifies the configuration to control the number of replicas for the created Deployment
+	// Replicas specifies the configuration to control the number of replicas for the created Deployment.
 	Replicas *GatewayClassReplicasConfig `json:"replicas,omitempty"`
-	// SecurityContext specifies the security context for the created Deployment's Pod
+	// SecurityContext specifies the security context for the created Deployment's Pod.
 	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
-	// Tolerations specifies the tolerations to use on the created Deployment
+	// Tolerations specifies the tolerations to use on the created Deployment.
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
-	// HostNetwork specifies whether the gateway pods should run on the host network
+	// HostNetwork specifies whether the gateway pods should run on the host network.
 	HostNetwork bool `json:"hostNetwork,omitempty"`
 	// TopologySpreadConstraints is a feature that controls how pods are spead across your topology.
 	// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/
 	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+	// DNSPolicy specifies the dns policy to use. These are set on a per pod basis.
+	DNSPolicy string `json:"dnsPolicy,omitempty"`
```

Comment:
Is there a reason to prefer `string` here over `corev1.DNSPolicy` with a `+kubebuilder:validation:Enum` like we do for `<GatewayClassServiceConfig>.Type`?

Reply:
none in particular, let me change this so we get that validation as well